src/rest-api/class-wp-rest-import-attachments.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+class WP_REST_Import_Attachments {
+	const NAMESPACE = 'wordpress-importer/v1';
+
+	function __construct() {
+		$this->register_routes();
+	}
+
+	protected function register_routes() {
+		register_rest_route( self::NAMESPACE, '/attachment', array(
+			// 'show_in_index' => false,
+			'methods' => WP_REST_Server::CREATABLE,
+			'callback' => array( $this, 'add_attachment' ),
+			'permission_callback' => array( $this, 'upload_and_import_permissions_check' ),
+			'args' => array(),
+		) );
+	}
+
+	function upload_and_import_permissions_check( $request ) {
+		// Begin copy from class-wp-rest-posts-controller
+		if ( ! empty( $request['id'] ) ) {
+			return new WP_Error( 'rest_post_exists', __( 'Cannot create existing post.' ), array( 'status' => 400 ) );
+		}
+
+		$post_type = get_post_type_object( 'attachment' );
+
+		if ( ! empty( $request['author'] ) && get_current_user_id() !== $request['author'] && ! current_user_can( $post_type->cap->edit_others_posts ) ) {
+			return new WP_Error( 'rest_cannot_edit_others', __( 'Sorry, you are not allowed to create posts as this user.' ), array( 'status' => rest_authorization_required_code() ) );
+		}
+
+		if ( ! empty( $request['sticky'] ) && ! current_user_can( $post_type->cap->edit_others_posts ) ) {
+			return new WP_Error( 'rest_cannot_assign_sticky', __( 'Sorry, you are not allowed to make posts sticky.' ), array( 'status' => rest_authorization_required_code() ) );
+		}
+
+		if ( ! current_user_can( $post_type->cap->create_posts ) ) {
+			return new WP_Error( 'rest_cannot_create', __( 'Sorry, you are not allowed to create posts as this user.' ), array( 'status' => rest_authorization_required_code() ) );
+		}
+
+		// Do we need this? probably not
+		// if ( ! $this->check_assign_terms_permission( $request ) ) {
+		//	return new WP_Error( 'rest_cannot_assign_term', __( 'Sorry, you are not allowed to assign the provided terms.' ), array( 'status' => rest_authorization_required_code() ) );
+		//	}
+
+		// End copy from class-wp-rest-posts-controller
```

Comment:
I know that this is here because `WP_REST_Attachments_Controller extends WP_REST_Posts_Controller` and it came from `WP_REST_Posts_Controller`... but do we need all of these checks for importing?

As far as I can tell `wp_import_handle_upload` (https://core.trac.wordpress.org/browser/tags/5.0.3/src/wp-admin/includes/import.php#L81) doesn't accept input for attachment id or author (for creating attachments as a different user), create a sticky attachment (if that's a thing), or do anything with terms... so it seems like those can be left out.

Checking for `create_posts`, `upload_files`, and `import` caps does make sense.

Reply:
Agreed -- we can pull out the `sticky`, `terms`, & `author` checks since we're not directly creating a post-like object.  Will make the change.